AK/Variant.h

```diff
@@ -40,47 +39,134 @@ consteval IndexType index_of()
     return VariantIndexOf<T, IndexType, 0, Ts...> {}();
 }
 
+template<typename... T>
+union VariantStorage;
+
+template<typename T, typename... Rest>
+union VariantStorage<T, Rest...> {
+    VariantStorage() = default;
+    VariantStorage()
+    requires(!IsTriviallyConstructible<T> || (!IsTriviallyConstructible<Rest> || ...)) {};
+
+    VariantStorage(VariantStorage&&) = default;
+    VariantStorage(VariantStorage const&) = default;
+
+    VariantStorage& operator=(VariantStorage&&) = default;
+    VariantStorage& operator=(VariantStorage const&) = default;
+
+    // Note: Actual destruction is handled by the Variant itself
+    ~VariantStorage() = default;
```

Comment:
I don't immediately see a reason to have a default overload that will be empty for trivially destructible types and an overload for the case of non-trivially destructible types that is explicitly empty but I suppose there's one, so a comment will be nice.

Reply:
IIRC think `=default` means something else than `{}`,
But sure I can add a comment explaining how and when a member is supposed to be constructed/destructed

Reply:
From https://stackoverflow.com/questions/56799129/when-to-make-a-destructor-defaulted-using-default:

> Versus `Implicit,` an explicitly defaulted destructor suppresses move operations, can be declared `private`, `protected`, or `noexcept(false)`, and in C++20 can be constrained (but not `consteval`). Very marginally, it can be declared `constexpr` to verify that it would be anyway. Declaring it `inline` doesn’t do anything. (It can also be out of line or virtual, but as stated above that can’t be a reason to use it.)

So there should be no reason for us to declare a defaulted destructor here?